R/transform-code.R

```diff
@@ -12,43 +12,50 @@ transform_code <- function(path, fun, verbose = FALSE, ...) {
     enc::transform_lines_enc(path, fun = fun, ..., verbose = verbose)
   } else if (is_rmd_file(path)) {
     enc::transform_lines_enc(path,
-      fun = partial(transform_rmd, transformer_fun = fun), ...,
+      fun = partial(transform_mixed, transformer_fun = fun, filetype = "Rmd"), ...,
+      verbose = verbose
+    )
+  } else if (is_rnw_file(path)) {
+    enc::transform_lines_enc(path,
+      fun = partial(transform_mixed, transformer_fun = fun, filetype = "Rnw"), ...,
       verbose = verbose
     )
   } else {
-    stop(path, " is not an R or Rmd file")
+    stop(path, " is not an R, Rmd or Rnw file")
   }
 }
 
-#' Transform Rmd contents
+#' Transform mixed contents
 #'
 #' Applies the supplied transformer function to code chunks identified within
-#' an Rmd file and recombines the resulting (styled) code chunks with the text
+#' an Rmd or Rnw file and recombines the resulting (styled) code chunks with the text
 #' chunks.
 #'
-#' @param lines A character vector of lines from an Rmd file
-#' @param transformer_fun A styler transformer function
+#' @param lines A character vector of lines from an Rmd or Rnw file.
+#' @param transformer_fun A styler transformer function.
+#' @param filetype A string indicating the filetype (Rmd or Rnw).
```

Comment:
I think here we could inherit `lines` and `filetype` from `separate_chunks()`.

Reply:
👍